File: domain/src/main/java/com/github/anrimian/musicplayer/domain/models/genres/Genre.java
package com.github.anrimian.musicplayer.domain.models.genres;

public class Genre {
    private final long id;

    private final String name;

    private final int compositionsCount;
    private final long totalDuration;

    public Genre(long id, String name, int compositionsCount, long totalDuration) {
        this.id = id;
        this.name = name;
        this.compositionsCount = compositionsCount;
        this.totalDuration = totalDuration;
    }

    public long getId() {
        return id;
    }

    public String getName() {
        return name;
    }

    public int getCompositionsCount() {
        return compositionsCount;
    }

    public long getTotalDuration() {
        return totalDuration;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        Genre genre = (Genre) o;

        return id == genre.id;
    }

    @Override
    public int hashCode() {
        return (int) (id ^ (id >>> 32));
    }

    @Override
    public String toString() {
        return "Genre{" +
                "id=" + id +
                ", name='" + name + '\'' +
                ", compositionsCount=" + compositionsCount +
                ", totalDuration=" + totalDuration +
                '}';
    }
}

File: domain/src/main/java/com/github/anrimian/musicplayer/domain/models/genres/Genre.kt
package com.github.anrimian.musicplayer.domain.models.genres

class Genre(
    val id: Long,
    val name: String,
    val compositionsCount: Int,
    val totalDuration: Long
) {

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as Genre

        if (id != other.id) return false

        return true
    }

    override fun hashCode(): Int {
        return id.hashCode()
    }

    override fun toString(): String {
        return "Genre(id=$id, name='$name', compositionsCount=$compositionsCount, totalDuration=$totalDuration)"
    }

}